Appsilon/shiny.gosling
# E2e tests

*Have you read the [Contributing Guidelines](https://github.com/Appsilon/.github/blob/main/CONTRIBUTING.md)?*

Closes #104 
Closes #68 

**What this PR is about**
* Only saving screenshots when running them from the cli `cypress run...` [as commented in this issue](https://github.com/cypress-io/cypress/issues/3324#issuecomment-542414532)
* Snapshot, visual tests using [cypress-plugin-snapshots](https://github.com/meinaart/cypress-plugin-snapshots)
* Added yarn scripts for every example app
* Added a wrapper script to run them all at once
* Added R scripts to run the Shiny apps


Added [new issue](https://github.com/Appsilon/shiny.gosling/issues/119)
* Remove TODOs from cypress tests - We need to find on-render'ish events for the SVG paths to use them when writing the snapshot tests. Right now we are using explicit `wait` statements which is not ideal as rendering it is not deterministic/depending on many factors affecting the time of render

### How to test
1. Read the [README](https://github.com/Appsilon/shiny.gosling/tree/e2e-tests/tests/cypress) on adding new tests
2. Ensure that you have your yarn dependencies updated (go to `tests/` and run `yarn install`)
3. Go to the `tests/` folder in the terminal and run `run_cypress_tests` -> Ensure that all of the tests have passed
4. Optional. You can inspect the `package.json` file and run some tests individually: for example via `yarn e2e-areaChart`
